openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx

```diff
@@ -65,6 +67,20 @@ const AlertDataInsightReportPage = () => {
     }
   }, []);
 
+  const handleSendDataInsightReport = async () => {
+    if (!isUndefined(dataInsightAlert)) {
+      try {
+        setIsSendingReport(true);
+        await triggerEventById(dataInsightAlert.id);
+        showSuccessToast(t('message.data-insight-report-send-success-message'));
+      } catch (error) {
+        showErrorToast(t('message.data-insight-report-send-failed-message'));
+      } finally {
+        setIsSendingReport(false);
+      }
+    }
```

Comment:
```suggestion

    if (isUndefined(dataInsightAlert)) {
      return;
    }
      try {
        setIsSendingReport(true);
        await triggerEventById(dataInsightAlert.id);
        showSuccessToast(t('message.data-insight-report-send-success-message'));
      } catch (error) {
        showErrorToast(t('message.data-insight-report-send-failed-message'));
      } finally {
        setIsSendingReport(false);
      }
    
```

Reply:
We are passing on the handler to button onClick and it expected return type void if we follow the above approach then the sonar cloud will flag it as a bug.